### PR TITLE
[ticket/39] Remove user api implementation and simplify switch

### LINKED ISSUE #39: Add switch for .com service

### PR DESCRIPTION
**Pull request by marc1706**
Fixed #39

### CODE REVIEW

File: acp/viglink_helper.php [hunk 1]
```diff
@@ -45,17 +45,45 @@ public function __construct(\phpbb\cache\service $cache, \phpbb\config\config $c
 	 */
 	public function set_viglink_services($force_update = false, $force_cache = false)
 	{
-		$versions = $this->get_versions_matching_stability($force_update, $force_cache);
+		$cache_file = '_versioncheck_viglink_' . $this->use_ssl;
```
Comment (michaelcullum) [inline acp/viglink_helper.php]: file?

File: acp/viglink_helper.php [hunk 2]
```diff
@@ -45,17 +45,45 @@ public function __construct(\phpbb\cache\service $cache, \phpbb\config\config $c
...
+		{
+			throw new \RuntimeException($this->user->lang('VERSIONCHECK_FAIL'));
+		}
+		else if ($info === false || $force_update)
+		{
+			try
+			{
+				$info = $this->file_downloader->get('www.phpbb.com', '/viglink', 'enabled', 443);
+			}
+			catch (\phpbb\exception\runtime_exception $exception)
```
Comment (michaelcullum) [inline acp/viglink_helper.php]: Can we catch any more specific exception here?

File: acp/viglink_helper.php [hunk 3]
```diff
@@ -45,17 +45,45 @@ public function __construct(\phpbb\cache\service $cache, \phpbb\config\config $c
...
+			{
+				$prepare_parameters = array_merge(array($exception->getMessage()), $exception->get_parameters());
+				throw new \RuntimeException(call_user_func_array(array($this->user, 'lang'), $prepare_parameters));
+			}
+			$error_string = $this->file_downloader->get_error_string();
+
+			if (!empty($error_string))
+			{
+				throw new \RuntimeException($error_string);
```
Comment (michaelcullum) [inline acp/viglink_helper.php]: We should allow error strings, for example a 500 or no network connection.

File: acp/viglink_helper.php (hunk 1 above)
Comment (marc1706) [inline acp/viglink_helper.php]: FWIW, that's what the version_helper says for some reason. Was just copy pasting parts of it.

File: acp/viglink_helper.php (hunk 2 above)
Comment (marc1706) [inline acp/viglink_helper.php]: Nope, that's exactly what it throws.

File: acp/viglink_helper.php [hunk 4]
```diff
@@ -45,17 +45,42 @@ public function __construct(\phpbb\cache\service $cache, \phpbb\config\config $c
...
-		array_map(array($this, 'set_viglink_configs'), $versions);
+			if ($info === '0')
+			{
+				$this->set_viglink_configs(array(
+					'allow_viglink_phpbb'	=> false,
+				));
+			}
+			else
+			{
+				$info = 1;
```
Comment (VSEphpbb) [inline acp/viglink_helper.php]: Should this be int or string (like '0' was a string above)?

File: acp/viglink_helper.php [hunk 5]
```diff
@@ -69,7 +94,6 @@ protected function set_viglink_configs($data)
 	{
 		$viglink_configs = array(
 			'allow_viglink_phpbb',
-			'allow_viglink_global',
```
Comment (VSEphpbb) [inline acp/viglink_helper.php]: I think you can delete the viglink.json test fixture

File: acp/viglink_module.php [hunk 6]
```diff
@@ -56,7 +56,6 @@ public function main($id, $mode)
 		// Get stored config/default values
 		$cfg_array = array(
 			'viglink_enabled' => isset($config['viglink_enabled']) ? $config['viglink_enabled'] : 0,
-			'viglink_api_key' => isset($config['viglink_api_key']) ? $config['viglink_api_key'] : '',
```
Comment (VSEphpbb) [inline acp/viglink_module.php]: Just as a side note, can be publicly decalre `tpl_name` and `page_title` and also replace `sizeof` with `count` in this module?

Comment (VSEphpbb) [inline acp/viglink_module.php]: This `viglink_api_key ` key can be removed from the listener tests

File: acp/viglink_module.php [hunk 7]
```diff
@@ -101,11 +93,7 @@ public function main($id, $mode)
 		}
 
 		// Set a general error message if VigLink has been disabled by phpBB
-		if (!$config['allow_viglink_global'])
-		{
-			$error[] = $language->lang('ACP_VIGLINK_DISABLED_GLOBAL');
```
Comment (VSEphpbb) [inline acp/viglink_module.php]: This language key can be removed from the lang file

File: migrations/viglink_data_v2.php [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+	public function effectively_installed()
+	{
+		return !isset($this->config['viglink_api_key']);
+	}
+
+	public function update_data()
+	{
+		return array(
+			// Basic config options
```
Comment (VSEphpbb) [inline migrations/viglink_data_v2.php]: We don't need this comment IMO

File: tests/event/acp_listener_test.php [hunk 9]
```diff
@@ -114,86 +114,6 @@ public function test_getSubscribedEvents()
 		), array_keys(\phpbb\viglink\event\acp_listener::getSubscribedEvents()));
 	}
 
-	/**
```
Comment (VSEphpbb) [inline tests/event/acp_listener_test.php]: Can you please add `\PHPUnit_Framework_MockObject_MockObject` to the var declarations for language, request and template vars

**Pull request merged**